featurebyte/query_graph/sql/interpreter/tile.py

```diff
@@ -116,35 +205,86 @@ def construct_tile_gen_sql(self, starting_node: Node) -> list[TileGenSql]:
                 event_table_timestamp_filter = self._get_event_table_timestamp_filter(groupby_node)
+                on_demand_entity_filters = None
 
-            info = self.make_one_tile_sql(groupby_node, event_table_timestamp_filter)
+            info = self.make_one_tile_sql(
+                groupby_node, event_table_timestamp_filter, on_demand_entity_filters
+            )
             sqls.append(info)
 
         return sqls
 
-    def _get_event_table_timestamp_filter(
-        self, groupby_node: GroupByNode
-    ) -> Optional[EventTableTimestampFilter]:
+    def _get_input_filter_context(self, groupby_node: GroupByNode) -> InputFilterContext:
         pruned_graph, node_name_map, _ = prune_query_graph(
             graph=self.query_graph, node=groupby_node
         )
 
-        # Check there is no lag operation, otherwise cannot apply timestamp filter directly
+        # Get node_types lineage
         pruned_node = pruned_graph.get_node_by_name(node_name_map[groupby_node.name])
+        node_types = set()
         for node in dfs_traversal(pruned_graph, pruned_node):
-            if node.type == NodeType.LAG:
-                return None
+            node_types.add(node.type)
 
-        # Identify which EventTable can be filtered. This is based on the timestamp column used in
-        # the groupby node.
+        # Get operation structure of the view before tile aggregation
         groupby_input_node = pruned_graph.get_node_by_name(
             pruned_graph.get_input_node_names(pruned_node)[0]
         )
         op_struct_info = OperationStructureExtractor(graph=pruned_graph).extract(groupby_input_node)
         op_struct = op_struct_info.operation_structure_map[groupby_input_node.name]
-        for column in op_struct.columns:
+
+        # Identify join keys lineage to propagate filtering to applicable tables
+        def _get_join_key_from_op_struct(
+            op_struct_obj: OperationStructure, column_name: str
+        ) -> Optional[SourceDataColumn]:
+            for column in op_struct_obj.columns:
+                if (
+                    column.name == column_name
+                    and isinstance(column, SourceDataColumn)
+                    and column.table_id is not None
+                ):
+                    return column  # type: ignore[return-value]
+            return None
+
+        join_keys_lineage = JoinKeysLineage()
+        for join_node in pruned_graph.iterate_nodes(pruned_node, NodeType.JOIN):
+            input_nodes = pruned_graph.get_input_node_names(join_node)
+            assert len(input_nodes) == 2
+            params = cast(JoinNodeParameters, join_node.parameters)
+            join_key_left = _get_join_key_from_op_struct(
+                op_struct_info.operation_structure_map[input_nodes[0]], params.left_on
+            )
+            join_key_right = _get_join_key_from_op_struct(
+                op_struct_info.operation_structure_map[input_nodes[1]], params.right_on
+            )
```

Comment:
Do we support joining tables on derived column? If yes, will this cause issue?

Reply:
We support joining tables on derived columns. Currently, `_get_join_key_from_op_struct` only retrieves `SourceDataColumn`. If the join is using derived columns as join keys this part won't fail, but the filter will not apply.

Reply:
If i understand this correctly, it means that it won't fail (just that the filter operation will not be applied before join). Since the main interest is to apply the filtering on the entity column, which should be source column, it this case, it should be fine.